src/store/EncryptedStore.ts

```diff
@@ -73,9 +73,31 @@ export default class EncryptedKeyStore implements KeyStore {
     // serialize the contents
     const bytes = bundle.encode()
     const wPreKey = getRandomValues(new Uint8Array(32))
-    const secret = hexToBytes(
-      await wallet.signMessage(storageSigRequestText(wPreKey))
-    )
+    const input = storageSigRequestText(wPreKey)
+    const walletAddr = await wallet.getAddress()
+
+    let sig = await wallet.signMessage(input)
+
+    // Check that the signature is correct, was created using the expected
+    // input, and retry if not. This mitigates a bug in interacting with
+    // LedgerLive for iOS, where the previous signature response is
+    // returned in some cases.
+    let address = utils.recoverAddress(utils.hashMessage(input), sig)
+    if (address !== walletAddr) {
+      sig = await wallet.signMessage(input)
+      console.log('invalid signature, retrying', address, walletAddr)
+    }
+
+    address = utils.recoverAddress(utils.hashMessage(input), sig)
+    if (address !== walletAddr) {
+      throw new Error('invalid signature')
+    }
```

Comment:
In this case I don't think lower level recoverAddress is needed. Verify message should work

```
const recoveredAddr = utils.verifyMessage(input, sig)
if ((await wallet.getAddress()) === recoveredAddr) {
  throw new Error('invalid storage signature')
}
```